Feature: Initial PWA Support (Service workers disabled)

translucentfocus opened the pull request:
### Reference to old Pull Request:  https://github.com/meteyou/mainsail/pull/80

Resolves [this issue](https://github.com/meteyou/mainsail/issues/60)

Utilized the Progressive Web Application (PWA) plugin https://cli.vuejs.org/core-plugins/pwa.html

Folder structure and img naming conventions follow the default pwa plugin format.  The favicon was moved per those standards

This commit includes the skeleton default service worker but I've disabled it, as I didn't see it being of great use (but could cause cache issues) to Mainsail until @meteyou wants to enable push notification.

Since the pwa plugin was added,  `package.json` and `package-lock.json` were both updated, so I might've dorked something up there.

I utilized the [current SVG logo from upstream master](https://github.com/meteyou/mainsail/blob/d6eb97ffe14686a1eac18c878ba5e0e616966654/src/assets/logo.svg) to create all .png assets. I can redo all of the icons once the new svg logo is updated.

### Lighthouse Results:
![image](https://user-images.githubusercontent.com/26986716/94352463-e5440a00-002a-11eb-9841-3c063d50b636.png)

### **Screenshots:**

Android Home Screen Pinning:
![Screenshot_20200926-185350](https://user-images.githubusercontent.com/26986716/94352467-073d8c80-002b-11eb-954e-a445d6409466.png)
![Screenshot_20200926-185401](https://user-images.githubusercontent.com/26986716/94352465-fbea6100-002a-11eb-81b2-522f3e3ed26a.png)

Android Splash Screen:
![Screenshot_20200926-185337](https://user-images.githubusercontent.com/26986716/94352494-63a0ac00-002b-11eb-908d-eedd186cb06b.png)

meteyou commented:
you cannot simply remove the favicon. there is a static link in App.vue. i change the favicon for print stat. pls check this at first.

translucentfocus commented:
> 
> 
> you cannot simply remove the favicon. there is a static link in App.vue. i change the favicon for print stat. pls check this at first.

Thank you for catching that! I updated the favicon references, (and fixed a broken path) & also changed the file paths to relative paths so that webpack can [map those in its dependency graph](https://cli.vuejs.org/guide/html-and-static-assets.html#relative-path-imports).

LazaroFilm commented:
I'll check it out next time I get to my printer!
